### PR TITLE
feat(media): save userId as tag on uploads

### PR DESCRIPTION
**Pull request by elbotho**

### CODE REVIEW

File: src/backend/media-route-handlers.ts [hunk 1]
```diff
@@ -90,7 +103,8 @@ export async function mediaPresignedUrl(req: Request, res: Response) {
     Bucket: bucketName,
     ContentType: mimeType,
     Metadata: { 'Content-Type': mimeType },
-    Tagging: `editorVariant=${editorVariant}&editorHost=${req.headers.host}`,
+    // saved as tags so we can potentially use it in IAM policies later:
+    Tagging: `editorVariant=${editorVariant}&editorHost=${editorHost}${userIdTag}`,
```
Comment (hugotiburtino) [inline src/backend/media-route-handlers.ts:107]: > editorHost=${editorHost}${userIdTag}`

Isn't it easier for a future search if we add a separator here? Something like `${editorHost}:${userIdTag}`

Comment (hugotiburtino) [reply src/backend/media-route-handlers.ts:107]: And a question here: Why isn't it in Metadata?

Comment (elbotho) [reply src/backend/media-route-handlers.ts:107]: > editorHost=${editorHost}${userIdTag}`
Isn't it easier for a future search if we add a separator here? Something like ${editorHost}:${userIdTag}

`userIdTag` is defined as `&userId=${userId}`

Comment (elbotho) [reply src/backend/media-route-handlers.ts:107]: > And a question here: Why isn't it in Metadata?

See comment above:
`    // saved as tags so we can potentially use it in IAM policies later:`

**Comment by elbotho** [comment]
(aside: currently tags are not actually saved in S3, I will make a follow up PR to fix that)

**Review by hugotiburtino — APPROVED** [review]

**Pull request merged**